session.go

```diff
@@ -396,8 +411,7 @@ func (s *Session) sendNoWait(hdr header) error {
 // send is a long running goroutine that sends data
 func (s *Session) send() {
```

Comment:
I meant to add a comment here in my review but I must have accidentally discarded it. Is it possible to just `defer close(s.sendDeadCh)` here?